util/jsonrpc-types/src/primitive.rs

```diff
@@ -50,3 +50,20 @@ pub type Timestamp = Uint64;
 ///
 /// This is a 32-bit unsigned integer type encoded as the 0x-prefixed hex string in JSON. See examples of [Uint32](type.Uint32.html#examples).
 pub type Version = Uint32;
+
+impl EpochNumberWithFraction {
```

Comment:
#### Why I disagree with this PR?

- `EpochNumberWithFraction` is an `alias` not a `new-type`.

  https://github.com/nervosnetwork/ckb/blob/e171a8a0ea7c1506547cb563fd8d8f3d267013b2/util/jsonrpc-types/src/primitive.rs#L28

  So `impl EpochNumberWithFraction` is equal to `impl Uint64`.

  Why `epoch_number`, `epoch_index` and `epoch_length` can be get from all `Uint64`?
  For example, developers will found `Capacity` has getters `epoch_number`, `epoch_index` and `epoch_length`, they will be confused.

- Bad smells: Duplicated Code and Magic Numbers 

  A better example:
  https://github.com/nervosnetwork/ckb/blob/e171a8a0ea7c1506547cb563fd8d8f3d267013b2/util/types/src/core/extras.rs#L412-L414

#### Let's look at this in another way:

- You could `impl From<Uint64> for ckb_types::EpochNumberWithFraction;`.
- 
- (Reuse Code) Then, get `epoch_number`, `epoch_index` or `epoch_length` from `ckb_types::EpochNumberWithFraction`.